### PR TITLE
fix(init): preserve settings.json symlink during atomic write

### LINKED ISSUE #653: "rtk init --global" breaks symlink
I share `settings.json` across multiple devices, but this command turned my symlink into a regular file.

I consider this a serious issue.

Another problem is that `settings.json` shouldn't contain absolute paths, and the patch content also includes absolute paths, making it impossible to share across multiple machines. I refactored them to relative paths.

As for why rtk.md is injected into CLAUDE.md, another issue already asked about it, so I won't ask here.

### PR DESCRIPTION
**Pull request by ChrisGVE**
## Summary

- `atomic_write()` replaced symlinked `settings.json` with a regular file, breaking setups that symlink config files (e.g. dotfile managers, multi-machine sync)
- Now resolves symlink targets before the tempfile+rename so the link itself is preserved

Closes #653

Supersedes #675 — rebased on current develop. The portable hook path change from #675 is no longer needed since the codebase moved from shell scripts to `rtk hook claude` binary command.

## Changes Made

- Added `resolve_atomic_target()` that detects symlinks and resolves to their target (absolute or relative)
- Updated `atomic_write()` to write through to the resolved target
- Added 2 unix-specific tests for absolute and relative symlink preservation

## Testing

- [x] `cargo fmt --all --check`
- [x] `cargo clippy --all-targets` — no warnings
- [x] `cargo test --all` — 1942 passed, 6 ignored
- [x] Specific tests: `cargo test init::tests::test_atomic_write`
- [x] Verified both fixes merge cleanly together (1947 tests pass combined)

## Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] No breaking changes introduced
- [x] Tests added for new functionality

### CODE REVIEW

**Comment by ChrisGVE** [comment]
@aeppling may I ask you to review this PR, this has been in the list for long and would help many user if it could be integrated into rtk. Thank you for your great work and for your time reviewing this PR.

File: src/hooks/init.rs [hunk 1]
```diff
@@ -382,13 +382,37 @@ fn write_if_changed(path: &Path, content: &str, name: &str, ctx: InitContext) ->
     }
 }
 
+/// Resolve the final write target: if `path` is a symlink, follow it so
+/// the atomic rename lands on the real file and the symlink is preserved.
+fn resolve_atomic_target(path: &Path) -> Result<PathBuf> {
```
Comment (KuSh) [inline src/hooks/init.rs]: Please use [fs::canonicalize](url) that handles all the hard things for you.

**Review by KuSh — CHANGES_REQUESTED** [review]

**Comment by KuSh** [comment]
Hi, thanks for your contribution!
As it is a rebase of someone else submission, you'll have to add "Co-authored-by" trailer to the commit's message.

File: src/hooks/init.rs (hunk 1 above)
Comment (ChrisGVE) [reply src/hooks/init.rs]: Done — replaced with `fs::canonicalize`, falls back to original path when file doesn't exist yet. All 3 `test_atomic_write*` tests pass (including both symlink variants).

**Comment by ChrisGVE** [comment]
Thanks @KuSh for the review! Both items addressed:

1. **`fs::canonicalize`** — replaced custom `resolve_atomic_target()` with `fs::canonicalize` + fallback for non-existent paths (new file creation). Handles chained symlinks, relative paths, all edge cases.

2. **Co-authored-by** — added `Co-authored-by: Roopesh <roopesh1724989@gmail.com>` trailer (original author of #675).

All tests green: `cargo fmt`, `cargo clippy`, `cargo test` — 1951 passed.

**Review by KuSh — APPROVED** [review]
LGTM, thanks
